Draft Blog announcement for 0.7.0 (#66)
kroxylicious/kroxylicious.github.io: _posts/2024-08-20-release-0_7_0.md

---
layout: post
title:  "Kroxylicious release 0.7.0"
date:   2024-08-20 00:00:00 +0000
author: "Keith Wall"
author_url: "https://www.github.com/k-wall"
categories:  [releases, kroxylicious, record-validation]
---

The Kroxylicious project is very pleased to announce the release of [Kroxylicious 0.7.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.7.0). See the [Changelog](https://github.com/kroxylicious/kroxylicious/blob/main/CHANGELOG.md#070) for a list of changes and summary of Deprecations, Changes and Removals.

### Record Validation 

> In this release, Kroxylcious dips its toes (or, should that be claws?) into the world of record validation.  You can now ensure that the content your producers send adheres to the schema you expect.   If a producer sends

Reviewer: ```suggestion
In this release, Kroxylicious dips its toes (or, should that be claws?) into the world of record validation.  You can now ensure that the content your producers send adheres to the schema you expect.   If a producer sends
```

> In this release, Kroxylcious dips its toes (or, should that be claws?) into the world of record validation.  You can now ensure that the content your producers send adheres to the schema you expect.   If a producer sends

Reviewer 2: > (or, should that be claws?)

Hehe nice 😁

> a message that doesn't meet the expected schema, the producer will get an error message and the message won't be delivered to the topic. Banish those poison messages from your topics for good!

Reviewer: ```suggestion
a record that doesn't meet the expected schema, the producer will get an error message and the record won't be delivered to the topic. Banish those poison records from your topics for good!
```

I found it a little tricky to read since we talk about message and error message in close proximity.

Reviewer 2: ```suggestion
a message that doesn't meet the expected schema, the producer will get an error response and the message won't be delivered to the topic. Banish those poison messages from your topics for good!
```

maybe we should use the Kafka term `record` rather than message?

Read more about the [use-case](https://kroxylicious.io/use-cases/#schema-validation-and-enforcement) or dive right into the [documentation](https://kroxylicious.io/docs/v0.7.0/#assembly-record-validation-filter-proxy).

> This is a preview feature and we are already thinking about enhancements. We'd love to hear your feedback. Thanks to [Carles Arnal](https://github.com/carlesarnal) for all the work he put into this feature.

Reviewer: Wondering if we should link to ApiCurio somewhere in here

Author: I couldn't quite find the right formulation of words to include Apicurio in the first couple of sentences whilst keeping things snappy.   Feel free to reword the announcement.   I cheated and added an apicurio label and a mention of Carles's project affiliation.

> 

Reviewer 2: I was starting to write up something along the lines of 
```suggestion
### Record Encryption plays nicely with bouncycastle
We had a request from the community to tweak how record encryption resolves its keys so that it was compatible with the [FIPS certified](https://www.bouncycastle.org/documentation/documentation-java/#bouncy-castle-java-fips-documentation) JCE provider [Bouncycastle](https://www.bouncycastle.org/). 
```

Mostly because FIPS seems to be a hot topic and showing we are responsive to community requests.
